add keys + functionality

Sharkules opened the pull request:
I've left the keys in the spawn room for testing, will put them in their correct locations once PR approved.

On pickup, keys increment playerKeys variable to track how many keys the player has found. Then the key script checks whether the player has enough keys, and if they do it sets the door to unlocked.

Sharkules committed:
add keys + functionality

Sharkules committed:
fix character crouch scaling - only effect controller

bchorne reviewed (changes requested):
So, the "Keys Remaining" UI can phase through walls, is that not drawing onto a canvas? It's fine if not, just give it the "Torch" Layer and it'll draw on the Overlay cam. Also, the "unlock" mechanism I was thinking of was leaving the main Valve set to inactive until all keys are collected, then having a dummy valve stand swap out for the real one. Bit tough to do programatically since I chucked two AN_Button components on the Valve object.

Sharkules committed:
fix ui, fix valve activation method

Sharkules commented:
Fixed keys ui phasing through walls
Changed keys collected visuals
Changed number of keys to 3
Added gameobject references for dummyvalve, realvalve and code that disables dummy and enables real once all keys collected.

Sharkules committed:
properly position keys, make ui text start at 0

bchorne reviewed (commented):
Good now

Sharkules committed:
change keys to prefab to avoid merge conflicts

Sharkules committed:
more prefabs
